main/src/main/scala/sbt/MainLoop.scala

```diff
@@ -147,7 +148,12 @@ object MainLoop {
     }
 
   def next(state: State): State = {
-    val context = LoggerContext(useLog4J = state.get(Keys.useLog4J.key).getOrElse(false))
+    val exchange = StandardMain.exchange
+    val taskTerminal = state.remainingCommands.headOption
+      .flatMap(_.source.flatMap(s => exchange.channelForName(s.channelName).map(_.terminal)))
+      .getOrElse(ITerminal.get)
+    val context =
+      LoggerContext(useLog4J = state.get(Keys.useLog4J.key).getOrElse(false), taskTerminal)
```

Comment:
+1 on this part of the code. Not sure about the name `taskTerminal` though. `channelTerminal`?